### PR TITLE
Add traces search

### PR DESCRIPTION
**Pull request by alonelaval**
related apache/skywalking#5273
11371597572573_ pic_hd
45141597584392_ pic_hd

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,4 @@
 [submodule "query-protocol"]
 	path = query-protocol
 	url = http://github.com/apache/skywalking-query-protocol/
+    branch = master
```
Comment (wu-sheng) [inline .gitmodules:4]: What is this?

Comment (alonelaval) [reply .gitmodules]: I haven't used submodules, ci errors, keep trying.

Comment (kezhenxu94) [reply .gitmodules]: > I haven't used submodules, ci errors, keep trying.

This should be reverted, it doesn't fix the CI, see my changes above

File: display/graph/tree/list.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+	} else {
+		noData := "no data"
+		list.Title = noData
+		tree.Title = noData
+	}
+
+	list.SetRect(0, 0, x, y)
+	tree.SetRect(x/4, 0, x, y)
+	help.SetRect(x-x/7, 0, x, y)
+	tree.ExpandAll()
```
Comment (kezhenxu94) [inline display/graph/tree/list.go:102]: The reason why we cannot switch the focus between the `list` and `tree` is that we manually set the rectangle area of them, by placing these widgets into containers, the focus can be switched between them, refer to `skywalking-cli/display/graph/linear/linear.go`

Comment (alonelaval) [reply display/graph/tree/list.go:102]: ok. i will check.

**Pull request merged**